lib/src/Uncertainty/Algorithm/Classification/MinimumVolumeClassifier.cxx

```diff
@@ -87,24 +100,34 @@ Indices MinimumVolumeClassifier::classify(const Sample & inS) const
   return ClassifierImplementation::classify(inS);
 }
 
-LevelSet MinimumVolumeClassifier::getLevelSet() const
+LevelSet MinimumVolumeClassifier::getLevelSet(const UnsignedInteger j) const
 {
   if (!isLevelSetComputed_)
   {
-    levelSet_ = distribution_.computeMinimumVolumeLevelSetWithThreshold(alpha_, threshold_);
+    levelSetCollection_.resize(alpha_.getSize());
+    threshold_.resize(alpha_.getSize());
+    for (UnsignedInteger i = 0; i < alpha_.getSize(); ++ i)
+      levelSetCollection_[i] = distribution_.computeMinimumVolumeLevelSetWithThreshold(alpha_[i], threshold_[i]);
```

Comment:
The `computeMinimumVolumeLevelSetWithThreshold()` should be complemented by a `computeMinimumVolumeLevelSetCollectionWithThreshold()` version taking two points as arguments, in order to speed-up computation e.g. when the levels are computed by sampling. The default implementation can easily be adapted:
```
Collection<LevelSet> DistributionImplementation::computeMinimumVolumeLevelSetCollectionWithThreshold(const Point & prob,
    Point & threshold) const
{
  if (!isContinuous()) throw NotYetImplementedException(HERE) << "In DistributionImplementation::computeMinimumVolumeLevelSet()";
  // 1D special case here to avoid a double construction of minimumVolumeLevelSetFunction
  if ((dimension_ == 1) && (ResourceMap::GetAsBool("Distribution-MinimumVolumeLevelSetBySampling")))
  {
    LOGINFO("Compute the minimum volume level set by sampling (QMC)");
    return result(computeUnivariateMinimumVolumeLevelSetCollectionByQMC(prob, threshold));
  }
  Function minimumVolumeLevelSetFunction(MinimumVolumeLevelSetEvaluation(clone()).clone());
  minimumVolumeLevelSetFunction.setGradient(MinimumVolumeLevelSetGradient(clone()).clone());
  // If dimension_ == 1 the threshold can be computed analyticaly
  Point minusLogPDFThreshold;
  const UnsignedInteger probSize = prob.getSize();
  if (dimension_ == 1)
  {
    const CompositeDistribution composite(minimumVolumeLevelSetFunction, *this);
    minusLogPDFThreshold = composite.computeQuantile(prob).asPoint();
    LOGINFO("Compute the minimum volume level set by using a composite distribution quantile (univariate general case)");
  } // dimension == 1
  else
  {
    LOGINFO("Compute the minimum volume level set by sampling (Monte Carlo)");
    const UnsignedInteger size = ResourceMap::GetAsUnsignedInteger("Distribution-MinimumVolumeLevelSetSamplingSize");
    const Sample xSample(getSample(size));
    const Sample logPDFSample(computeLogPDF(xSample));
    minusLogPDFThreshold = -logPDFSample.computeQuantile(Point(probSize, 1.0) - prob).asPoint();
  } // dimension > 1
  threshold = Point(probSize);
  Collection<LevelSet> levelSets(probSize);
  for (UnsignedInteger i = 0; i < probSize; ++i)
  {
    threshold[i] = std::exp(-minusLogPDFThreshold[i]);
    levelSets[i] =  LevelSet(minimumVolumeLevelSetFunction, LessOrEqual(), minusLogPDFThreshold[i]);
  }
  return levelSets;
}
```
This way, the costly computeLogPDF() samples are reused for all the levels. Note that you have to implement a `computeUnivariateMinimumVolumeLevelSetCollectionByQMC()` method the same way the Monte Carlo version is computed. For the few distributions with a specific implementation of the `computeMinimumVolumeLevelSetWithThreshold()` method (EllipticalDistribution, CompositeDistribution, Frechet, GeneralizedPareto, IndependentCopula, InverseNormal, ParametrizedDistribution, Pareto, RandomMixture, TruncatedNormal, Uniform) some additional work is needed, but not so much.

Reply:
I propose to postpone this change to another pr if that's ok for you, I've created #1734 for this